ibis/omniscidb/tests/conftest.py

```diff
@@ -36,26 +36,70 @@ def con():
     )
 
 
+class temp_helper:
+    def __init__(
+        self,
+        con,
+        kind,
+        create=False,
+        create_kwargs=dict(),
+        extra_return=[],
+        method_name=None,
+    ):
+        self.name = _random_identifier(kind)
+        self.con = con
+        self.kind = kind
+        self.create = create
+        self.create_kwargs = create_kwargs
+        self.extra_return = extra_return
+        self.method_name = kind if method_name is None else method_name
+
+    def __enter__(self):
+        # some of the temp entities may not support 'force' parameter
+        # at 'drop' method, that's why we use 'try-except' for that
+        try:
+            getattr(self.con, 'drop_' + self.method_name)(self.name)
+        except Exception:
+            pass
+
+        if self.create:
+            getattr(self.con, 'create_' + self.method_name)(
+                self.name, **self.create_kwargs
+            )
+        return self.name
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        try:
+            getattr(self.con, 'drop_' + self.method_name)(self.name)
+        except Exception:
+            pass
+
+
 @pytest.fixture(scope='function')
-def test_table(con):
+def test_schema() -> ibis.schema:
     """
-    Define fixture for test table.
+    Define fixture for test schema
 
-    Yields
+    Returns
     -------
-    ibis.expr.types.TableExpr
+    ibis.expr.Schema
     """
-    table_name = 'test_table'
-    con.drop_table(table_name, force=True)
-
-    schema = ibis.schema(
+    return ibis.schema(
         [('a', 'polygon'), ('b', 'point'), ('c', 'int8'), ('d', 'double')]
     )
-    con.create_table(table_name, schema=schema)
 
-    yield con.table(table_name)
 
-    con.drop_table(table_name)
+@pytest.fixture(scope='function')
+def test_table(con, test_schema):
+    """
+    Define fixture for test table.
+
+    Yields
+    -------
+    ibis.expr.types.TableExpr
+    """
+    with temp_helper(con, 'table', True, {'schema': test_schema}) as name:
```

Comment:
For things that are not obvious, I think it's better to use the parameter name. Reading this, it's difficult to guess what's `True` without having to go to the function definintion. I think being explicit helps readability:
```suggestion
    with temp_helper(con, kind='table', create=True, create_kwargs={'schema': test_schema}) as name:
```

Reply:
done